### PR TITLE
Feature/300 closed loop dv scenario

### PR DESCRIPTION
**Pull request by JulianHammerl**
* **Tickets addressed:** Confluence 300
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

A BSK-Sim scenario was created that illustrates the the closed loop performance of the dv guidance modules (attitude control and maneuver execution) for Delta-V maneuvers.
The commits are organized the following way:

- Commits 1-5: from full dv guidance scenario (illustrating Lambert modules)
- Commits 6-8: Update BSK-Sim support files (BSK_Dynamics, BSK_Fsw and BSK_Plotting)
- Commit 9: Add BSK-Sim scenario
- Commit 10: Update BSK release notes

Reviewers should be aware that this branch is based off of the Ticket 270 branch (#122  ), so the #122 PR should be merged first.

## Verification

All modules used by this scenario include their own unit tests, so no test was included for this scenario. Only the dvExecuteGuidance does not include a unit test, but a future ticket was created to add a unit test for that module. 

## Documentation

Nothing was invalidated.

## Future work

The thrFiringRemainder and dvExecuteGuidance modules will be refactored.

### CODE REVIEW

File: examples/BskSim/scenarios/scenario_LambertGuidance.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    TheScenario = scenario_LambertGuidance()
+    runScenario(TheScenario)
+    figureList = TheScenario.pull_outputs(showPlots)
+
+    return figureList
+
+
+
+
+
```
Comment (thibaudteil) [inline examples/BskSim/scenarios/scenario_LambertGuidance.py]: extra lines?

File: src/fswAlgorithms/orbitControl/lambertPlanner/lambertPlanner.cpp [hunk 2]
```diff
@@ -44,10 +44,10 @@ void LambertPlanner::Reset(uint64_t currentSimNanos)
     }
 
     // check that the provided input module parameters are valid
-    if (this->mu <= 0.0){
+    if (this->mu < 0.0){
         bskLogger.bskLog(BSK_ERROR, "lambertPlanner: mu must be positive.");
     }
-    if (this->finalTime - this->maneuverTime <= 0.0){
+    if (this->finalTime - this->maneuverTime < 0.0){
         bskLogger.bskLog(BSK_ERROR,
```
Comment (thibaudteil) [inline src/fswAlgorithms/orbitControl/lambertPlanner/lambertPlanner.cpp]: You'll probably have to watch merge conflicts between this and the other branch

File: src/fswAlgorithms/orbitControl/lambertValidator/lambertValidator.cpp [hunk 3]
```diff
@@ -57,7 +57,7 @@ void LambertValidator::Reset(uint64_t currentSimNanos)
     }
 
     // check that the provided input module parameters are valid
```
Comment (thibaudteil) [inline src/fswAlgorithms/orbitControl/lambertValidator/lambertValidator.cpp]: Same here, just fyi

**Review by thibaudteil — COMMENTED** [review]
Looks good

File: examples/BskSim/scenarios/scenario_LambertGuidance.py (hunk 1 above)
Comment (JulianHammerl) [reply examples/BskSim/scenarios/scenario_LambertGuidance.py]: Fixed on the 270-full-dv-guidance-scenario branch (this PR is branched off 270).

File: src/fswAlgorithms/orbitControl/lambertPlanner/lambertPlanner.cpp (hunk 2 above)
Comment (JulianHammerl) [reply src/fswAlgorithms/orbitControl/lambertPlanner/lambertPlanner.cpp]: Noted.

**Review by thibaudteil — APPROVED** [review]

**Pull request merged**